### ISSUE TITLE
multiple deployments

### ISSUE THREAD

**Issue by laurenskling** [crgeary/wp-jamstack-deployments#44]
So one of my projects depending on this awesome plugin is now using a test deployment to check changes before going live. It would be super powerful and useful if the Wordpress could have 2 (or more) buttons: Deploy to X, Deploy to Y.

Would this be something we could achieve?

**Comment by crgeary** [comment]
How would you envisage this working when you save a post/page in WordPress? As this plugin will trigger deployments when content is changed, I'm not sure how we would determine which deployment to trigger.

**Comment by laurenskling** [comment]
Well, it only triggers automatically if you have checked anything. If you dont have anything checked its a manual deploy via the button only. This is how im working production now.

If the settings was an array of the settings, you could have one with checkmarks for everything, going to test, and one endpoint with nothing checked to do manual production deploys

**Comment by rv-bnowak** [comment]
I greatly second @laurenskling 's ask! This would be a great feature to have. I could have one "element" of settings in the mentioned array that related to production, one that related to our staging (or other branch) environments. That would be awesome!

**Comment by laurenskling** [comment]
Not only would it be great for production/staging, I'm building a small Gatsby site from a big WP now (only the blog posts); but if i'd want to make another Gatsby site with content from this WP site, I won't have a way to deploy it.

**Comment by satoshionoda** [comment]
I’m third for this!

For the UI matter, I guess something like [Custom Post Type UI](https://wordpress.org/plugins/custom-post-type-ui/) would suit this function. Though it’s going to be a huge revamp of the plugin.

Meantime, as I seriously needed for the staging environment, I’ve cloned this plugin and just modified variable names and keys so this modified version can be installed to next to the original. I haven’t fully tested yet, it works as I expected so far. 
https://github.com/satoshionoda/wp-jamstack-deployments

@laurenskling @rv-bnowak 
Feel free to use it at your own risk.

@crgeary 
I suppose GPL allows modification and redistribution but if you don’t like the idea, please let me know. I’ll close the repository and use it just for me.

**Comment by crgeary** [comment]
@satoshionoda I don't mind at all. The only reason I made this GPL, is because of WordPress. Otherwise I would have made it MIT.

It's unfortunately low on my priority list, so I'd rather see you offer the feature for people who need it, even if it's not part of my own plugin. That's fine with me.

**Comment by CHEWX** [comment]
Yeah would be good to see this, definitely have a use case for staging for editors to get sign off on work.

**Comment by laurenskling** [comment]
I actually need 3 deployments from my wordpress now. I'm currently webooking a system that calls the 3 webooks 🤷 

Would be awesome if we could get something to work from within this plugin, still really loving this plugin a bunch. Thanks so much for making it
